### PR TITLE
Draft: Up-to-date with upstream

### PR DESCRIPTION
**Pull request by mauneyc-LANL**
## PR Summary

Collect updates from spiner and ports-of-call.

The `FetchContent` code for `spiner` was slightly modified and used to replace the `singularity_import_*` code; this should make dependencies handle more consistently across codes. 

@jhp-lanl your fixes in #220 will be overwritten, but I still need to clear up some build issues on my end, so that isn't going to interfere.

2023-02-03: depends on https://github.com/lanl/spiner/pull/58

## One Must Imagine Sisyphus Happy

- [x] ~~`spiner` builds tests even with `BUILD_TESTING=OFF` cache set. why?~~ a3405a4
- [ ] with `kokkos+kokkoskernels` (openmp backend) segfaults on `pte` test
- [ ] some CMake lists have duplicate entries - these shouldn't break anything but why take the chance
- [ ] Check if "all-`Fetch`" build works (no spack, except hdf5)
- [ ] Check some mixed builds
- [ ] GPU build works
- [ ] make sure github/gitlab CI is up2date

## PR Checklist

- [ ] Adds a test for any bugs fixed. Adds tests for new features.
- [ ] Format your changes by using the `make format` command after configuring with `cmake`.
- [ ] Document any new features, update documentation for changes made.
- [ ] Make sure the copyright notice on any files you modified is up to date.
- [ ] After creating a pull request, note it in the CHANGELOG.md file
- [ ] If preparing for a new release, update the version in cmake.

### CODE REVIEW

**Comment by dholladay00** [comment]
Are the naming changes to reduce probability of collisions @mauneyc-LANL ?

File: CMakeLists.txt [hunk 1]
```diff
@@ -242,75 +268,63 @@ message(STATUS "Portability strategy set to ${PORTABILITY_STRATEGY}")
...
 if (SINGULARITY_BUILD_CLOSURE)
   if(SINGULARITY_USE_KOKKOSKERNELS)
-    # type: external
-    singularity_import_dependency(
-      PKG KokkosKernels
-      TARGETS Kokkos::kokkoskernels
-      )
+    # kokkoskernels solvers
+    # Disable TPLs
+    list(APPEND _kk_tpls "BLAS" "MKL" "LAPACK" "CUBLAS" "CUPARSE" "MAGMA")
```
Comment (dholladay00) [inline CMakeLists.txt:273]: This is brilliant! 

It won't collide with an externally installed kokkos kernels correct?

**Comment by dholladay00** [comment]
@mauneyc-LANL what's the timeline for merging this, are we waiting for other MRs in other repos?

**Comment by mauneyc-LANL** [comment]
> @mauneyc-LANL what's the timeline for merging this, are we waiting for other MRs in other repos?

I am not sure about the usefulness of several significant changes in this PR. I'm thinking to close this and refactor only the minimum changes to stay flush with upstream.

**Comment by mauneyc-LANL** [comment]
Closing, will resubmit with simpler changes.

**Pull request closed without merging**